### ISSUE TITLE
NPE when using mock Kubernetes Server

### ISSUE THREAD

**Issue by lalithsuresh** [fabric8io/kubernetes-client#1818]
Following what I could find in kubernetes-tests, I'm running into the following NPE when using the Kubernetes mock server.

```java
public class MyTest {
    @Rule
    public KubernetesServer server = new KubernetesServer(false);

    @Test
    public void testInformer() {
        System.out.println(server.expect());
    }
}
```

NPE:
```
java.lang.NullPointerException
	at io.fabric8.kubernetes.client.server.mock.KubernetesServer.expect(KubernetesServer.java:72)
```

The dependencies I used were:

```xml
         <dependency>
            <groupId>io.fabric8</groupId>
            <artifactId>kubernetes-client</artifactId>
            <version>4.6.1</version>
        </dependency>
        <dependency>
            <groupId>io.fabric8</groupId>
            <artifactId>kubernetes-server-mock</artifactId>
            <version>4.6.1</version>
            <scope>test</scope>
        </dependency>   
```

I've tried different variants of the KubernetesServer constructors to no avail. What am I missing?

**Comment by lalithsuresh** [comment]
It appears that server.before() isn't being invoked by the @Rule. Invoking it manually works.

**Comment by rohanKanojia** [comment]
@lalithsuresh : ah, I see. Would it be possible for you to raise a PR to fix this behavior(if you have some time ;-) )?

**Comment by lalithsuresh** [comment]
I'll have a look sometime in the coming days.

**Comment by SahibYar** [comment]
This issue should not be marked as closed / completed, facing same issue in `6.11.0`
